crates/analyzer/src/namespace/items.rs

```diff
@@ -1891,6 +1878,11 @@ impl TraitId {
     }
 
     pub fn is_implemented_for(&self, db: &dyn AnalyzerDb, ty: TypeId) -> bool {
+        // All structs automagically implement the Emittable trait
+        if self.is_std_trait(db, EMITTABLE_TRAIT_NAME) && ty.is_struct(db) {
+            return true;
+        }
```

Comment:
This breaks the invariant of a group of methods related to trait implementation because this function returns `true` even if the actual implementation doesn't exist. So all other codes which depend on the method would need additional checks to ensure the actual implementation exists.

Reply:
>even if the actual implementation doesn't exist

Can you elaborate on that? The method is implemented by the compiler (compiler magic basically) for any struct (well, maybe we need to restrict it a bit more because as you hinted structs with enums should maybe not implement `Emittable`). For any `Emittable` the compiler ensures that `emit` can be called.

Reply:
If the method is properly implemented by the compiler,  then why are these awkward lines necessary?
Do these types have `ImplId` for the `Emittable` trait?  I'd like to expect that `ImplId` should exist when the `is_implemented_for` returns `true`.